scripts/prepare-github-release.mjs

```diff
@@ -0,0 +1,89 @@
+// @ts-check
+
+import { appendFile } from "node:fs/promises";
+import { readFile } from "node:fs/promises";
+
+import { getLatestPackageVersionFromNpm, readPackage } from "./lib/packages.mjs";
+
+// The regex was taken from https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+const semverRegex = /^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/;
+
+function isPrerelease(version) {
+  const [, , , , prerelease] = version.match(semverRegex);
+  return prerelease !== undefined;
+}
+
+async function isLatest(version) {
+  const latestVersion = await getLatestPackageVersionFromNpm("hardhat");
+  return version === latestVersion;
+}
+
+async function getHardhatChangelogEntry(version) {
+  const changelog = await readFile("v-next/hardhat/CHANGELOG.md", "utf-8");
+
+  const lines = changelog.split('\n')
+  const headerIndex = lines.findIndex((line) => line == `## ${version}`)
+
+  if (headerIndex == -1) {
+    throw new Error(`Changelog entry for version ${version} not found in ./v-next/hardhat/CHANGELOG.md`)
+  }
+
+  const entryLines = [];
+
+  for (const line of lines.slice(headerIndex + 1)) {
+    if (line.startsWith('## ')) {
+      break
+    }
+    if (line === '### Patch Changes') {
```

Comment:
This line we will need to be careful of when we shift to the full release process for 3.0.0. But the entire process will need reviewed for that.

I think I will add an issue to indicate we need to review and completely upgrade these scripts for `3.0.0`:

https://github.com/NomicFoundation/hardhat/issues/6693